src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PlatformCompatibilityAnalyzer.cs

```diff
@@ -1509,43 +1497,66 @@ static void MergePlatformAttributes(ImmutableArray<AttributeData> immediateAttri
                                     }
                                 }
                             }
-                            // other platform attributes are ignored as the list couldn't be extended
+                            else // not existing parent platforms might need to be removed
+                            {
+                                notFoundPlatforms.Add(platform);
+                            }
+                        }
+                    }
+                    // For allow list if child narrowing supported platfroms by having less platforms support than parent,
+                    // not existing parent platforms should be removed
+                    if (notFoundPlatforms.Count > 0)
+                    {
+                        if (supportFound)
+                        {
+                            childAttributes = new SmallDictionary<string, Versions>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Because `SmallDictionary` doesn't have the `Remove` method added this temporary workaround, planning to add remove method to the `SmallDictionary`, then this will be updated